### DISCUSSION TITLE
Path parameter redefines built-in

### DISCUSSION THREAD

**Discussion by alexschoeberl** [fastapi/fastapi#6654] [Questions] [answered]
I would like to use a path parameter with the name "id". However, it is not the best idea to use "id" as a function argument, since it redefines a built-in name. Also pylint complains about this.
Is there a possibility to have the path parameter and the function argument use different names? For example id and id_?

**Comment by Kludex** [comment, marked as answer]
```python
from fastapi import FastAPI, Path

app = FastAPI()

@app.get("/{id}")
def something(id_: int = Path(..., alias="id")):
    ...
```
